src/ui/public/timepicker/kbn_global_timepicker.js

```diff
@@ -25,6 +26,73 @@ UiModules
       $rootScope.toggleRefresh = () => {
         timefilter.refreshInterval.pause = !timefilter.refreshInterval.pause;
       };
+
+      // functions for the time navigation buttons
+
+      // travel forward in time based on the interval between from and to
+      $scope.foward = function () {
+        const time = getFromTo();
+        const diff = time.to.diff(time.from);
+        const origTo = time.to.toISOString();
+
+        time.to.add(diff, 'milliseconds');
+        timefilter.time.from = origTo;
+        timefilter.time.to = time.to.toISOString();
+      };
+
+      // travel backwards in time based on the interval between from and to
+      $scope.back = function () {
+        const time = getFromTo();
+        const diff = time.to.diff(time.from);
+        const origFrom = time.from.toISOString();
+
+        time.from.subtract(diff, 'milliseconds');
+        timefilter.time.to = origFrom;
+        timefilter.time.from = time.from.toISOString();
+      };
+
+      // zoom out, doubling the difference between start and end, keeping the same time range center
+      $scope.zoomOut = function () {
+        const time = getFromTo();
+        const from = time.from.unix() * 1000;
+        const to = time.to.unix() * 1000;
+
+        const diff = Math.floor((to - from) / 2);
+
+        timefilter.time.from = moment(from - diff).toISOString();
+        timefilter.time.to = moment(to + diff).toISOString();
+      };
+
+      // zoom in, halving the difference between start and end, keeping the same time range center
+      $scope.zoomIn = function () {
+        const time = getFromTo();
+        const from = time.from.unix() * 1000;
+        const to = time.to.unix() * 1000;
+
+        const diff = Math.floor((to - from) / 4);
+
+        timefilter.time.from = moment(from + diff).toISOString();
+        timefilter.time.to = moment(to - diff).toISOString();
+      };
+
+      // find the from and to values from the timefilter
+      // if a quick or relative mode has been selected, work out the
+      // absolute times and then change the mode to absolute
+      function getFromTo() {
+        if (timefilter.time.mode === 'absolute') {
+          return {
+            to:   moment(timefilter.time.to),
+            from: moment(timefilter.time.from)
+          };
+        } else {
+          timefilter.time.mode = 'absolute';
```

Comment:
minor: consider not doing the side-effect in this getter/setter. If you prefer to avoid two function-calls, perhaps rename this function to something like `setToTimeToAbsoluteAndGetToFrom` (yeah, i know :$). But now, this state change is a little buried.